### ISSUE TITLE
since argument does not work work in binance fetch_trades futures

### ISSUE THREAD

**Issue by kirilldanilenko89** [ccxt/ccxt#12127]
Hi,

Using since argument in fetch_trades does not working when fetching futures Binance trades. Its always return recent 500 trades. Works ok with params={'fromId':previous_trade_id} but not with timestamp. The same code but with spot option works ok with since. 

CCTX version ccxt==1.74.40

Code:

`
 import os
 import sys
 import csv

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(root + '/python')

import ccxt  # noqa: E402

exchange = ccxt.binance({
    'enableRateLimit': True,
    'options': {
        'defaultType': 'future',
    }
})

markets = exchange.load_markets()
symbol = 'ICX/USDT'
market = exchange.market(symbol)
one_hour = 3600 * 1000
since = exchange.parse8601('2022-02-26T00:00:00')
now = exchange.milliseconds()
#previous_trade_id = 45138270
previous_trade_id = None
filename = exchange.id + '_' + market['id'] + '.csv'

cnt = 0

with open(filename, mode="w") as csv_f:
    csv_writer = csv.DictWriter(csv_f, delimiter=",", fieldnames=["id", "timestamp", "size", "price", "side"])
    csv_writer.writeheader()
    print('Start... since {0} till {1}'.format(since, now))
    while since < now:
        try:
            if previous_trade_id != None:
                trades = exchange.fetch_trades(symbol, params={'fromId':previous_trade_id + 1})
            else:
                trades = exchange.fetch_trades(symbol, since)

            if len(trades) > 0:
                previous_trade_id = int(trades[-1]['id'])
            else:
                break

            print("last trade id {}".format(previous_trade_id))

            for trade in trades:
                csv_writer.writerow({
                    'id': trade['id'],
                    'timestamp': trade['timestamp'],
                    'size': trade['amount'],
                    'price': trade['price'],
                    'side': trade['side']
                })

            cnt += len(trades)

        except ccxt.NetworkError as e:
            print(type(e).__name__, str(e))
            exchange.sleep(60000)

    print(cnt)
`

**Comment by ttodua** [comment]
To start with:
- please use code-block to correctly post your code. if you look at your code now - it is not readable.
- you are setting since only once to '2022-02-26T00:00:00' and of course, when you call your code, it will have more than 500 trades after that time. you should re-set the 'since' parameter after calling, like:
```
        if len(trades) > 0:
            previous_trade_id = int(trades[-1]['id'])
            since = int(trades[-1]['timestamp'])
        else:
            break
```

then you can use that 'since' timestamp, so next call `fetch_trades` will use that `since` timestamp. I don't understand how it works for spot - it might be because there were less trades in spot compared to futures , that might be reason you got less trades in one of them.

**Comment by kirilldanilenko89** [comment]
Thanks for answer. I simplified the code.

```
import os
import sys

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(root + '/python')

import ccxt  # noqa: E402

exchange = ccxt.binance({
    'enableRateLimit': True,
    'options': {
        'defaultType': 'future',
    }
})

markets = exchange.load_markets()
symbol = 'ICX/USDT'
market = exchange.market(symbol)
since = exchange.parse8601('2022-02-26T00:00:00')

trades = exchange.fetch_trades(symbol, since)

print('since {}\nfetch {}'.format(since, trades[0]['timestamp']))
```

Output:

```
since 1645833600000
fetch 1646003101847
``` 

If i change defaultType to 'spot' fetch_trade works correct.

Output for spot

```
since 1645833600000
fetch 1645833602201
```

Seems fetch_trade with option futures return recent 500 trades,   not from since.

CCTX 1.74.40
MacOS
python 3.9

**Comment by kroitor** [comment]
@kirilldanilenko89 thanks for reporting, fixed this issue in CCXT 1.74.44+ (the new build will arrive in ~20 minutes).

**Comment by kirilldanilenko89** [comment]
version 1.74.44 works.thanks!
